### ISSUE TITLE
Enums are not handled correctly when they are used as map keys

### ISSUE THREAD

**Issue by pwinckles** [avaje/avaje-jsonb#131]
Example:

```java
import io.avaje.jsonb.Json;
import io.avaje.jsonb.Jsonb;

import java.util.Map;

@Json
public record EnumExample(String name, Map<Thing, String> thingMap) {

    public enum Thing {
        ONE,
        TWO
    }

    public static void main(String[] args) {
        var jsonb = Jsonb.builder().build();
        var type = jsonb.type(EnumExample.class);
        var example = new EnumExample("test", Map.of(
                EnumExample.Thing.ONE, "1",
                EnumExample.Thing.TWO, "2"
        ));
        System.out.println(type.toJson(example));
    }

}
```

That will produce an exception like:

```
Exception in thread "main" io.avaje.jsonb.JsonException: java.lang.ClassCastException: class com.pwinckles.EnumExample$Thing cannot be cast to class java.lang.String (com.pwinckles.EnumExample$Thing is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')
	at io.avaje.jsonb.core.DJsonType.toJson(DJsonType.java:89)
	at io.avaje.jsonb.core.DJsonType.toJson(DJsonType.java:61)
	at com.pwinckles.EnumExample.main(EnumExample.java:23)
Caused by: java.lang.ClassCastException: class com.pwinckles.EnumExample$Thing cannot be cast to class java.lang.String (com.pwinckles.EnumExample$Thing is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')
	at io.avaje.jsonb.core.MapAdapter.toJson(MapAdapter.java:51)
	at io.avaje.jsonb.core.MapAdapter.toJson(MapAdapter.java:27)
	at io.avaje.jsonb.NullSafeAdapter.toJson(NullSafeAdapter.java:33)
	at com.pwinckles.jsonb.EnumExampleJsonAdapter.toJson(EnumExampleJsonAdapter.java:52)
	at com.pwinckles.jsonb.EnumExampleJsonAdapter.toJson(EnumExampleJsonAdapter.java:11)
	at io.avaje.jsonb.NullSafeAdapter.toJson(NullSafeAdapter.java:33)
	at io.avaje.jsonb.core.DJsonType.toJson(DJsonType.java:86)
	... 2 more
```

**Comment by SentryMan** [comment]
it seems it defaults to only processing Map<String, whatever>, I'll take a look.

**Comment by SentryMan** [comment]
Alright, so got a fix and waiting for it to deploy. 

If you don't wanna wait, you can add this file to your code and it shall work.

```
@CustomAdapter
@SuppressWarnings("unchecked")
final class EnumMapAdapter<K extends Enum<K>, V> implements JsonAdapter<Map<K, V>> {

  static final Factory FACTORY =
      (type, jsonb) -> {
        final Class<?> rawType = Util.rawType(type);
        if (rawType != EnumMap.class && rawType != Map.class) {
          return null;
        }
        final var types = Util.mapValueTypes(type, rawType);
        if (!((Class<?>) types[0]).isEnum()) {
          return null;
        }
        return new EnumMapAdapter<>(jsonb, types).nullSafe();
      };
  private final Class<K> enumClass;
  private final JsonAdapter<V> valueAdapter;

  EnumMapAdapter(Jsonb jsonb, Type[] valueType) {
    this.enumClass = (Class<K>) valueType[0];
    this.valueAdapter = jsonb.adapter(valueType[1]);
  }

  @Override
  public void toJson(JsonWriter writer, Map<K, V> map) {
    writer.beginObject();
    for (final var entry : map.entrySet()) {
      if (entry.getKey() == null) {
        throw new JsonDataException("Map key is null at " + writer.path());
      }
      writer.name(entry.getKey().name());
      valueAdapter.toJson(writer, entry.getValue());
    }
    writer.endObject();
  }

  @Override
  public Map<K, V> fromJson(JsonReader reader) {
    final Map<K, V> result = new EnumMap<>(enumClass);
    reader.beginObject();
    while (reader.hasNextField()) {
      final String name = reader.nextField();
      final var enumVal = Enum.valueOf(enumClass, name);
      final V value = valueAdapter.fromJson(reader);
      final V replaced = result.put(enumVal, value);
      if (replaced != null) {
        throw new JsonDataException(
            String.format(
                "Map key '%s' has multiple values at path %s : %s and %s",
                name, reader.location(), replaced, value));
      }
    }
    reader.endObject();
    return result;
  }
}
```

**Comment by rbygrave** [comment]
FYI: The fix is in 1.6-RC6 if you want to try that out.

**Comment by pwinckles** [comment]
Thanks! I confirmed that it's behaving as expected now.
